tests/integration/blockchain/test_currency.py

```diff
@@ -0,0 +1,117 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import pytest
+from decimal import Decimal, InvalidOperation
+
+from nucypher.blockchain.eth.token import NU
+
+
+def test_NU(token_economics):
+
+    # Starting Small
+    min_allowed_locked = NU(token_economics.minimum_allowed_locked, 'NuNit')
+    assert token_economics.minimum_allowed_locked == int(min_allowed_locked.to_nunits())
+
+    min_NU_locked = int(str(token_economics.minimum_allowed_locked)[0:-18])
+    expected = NU(min_NU_locked, 'NU')
+    assert min_allowed_locked == expected
+
+    # Starting Big
+    min_allowed_locked = NU(min_NU_locked, 'NU')
+    assert token_economics.minimum_allowed_locked == int(min_allowed_locked)
+    assert token_economics.minimum_allowed_locked == int(min_allowed_locked.to_nunits())
+    assert str(min_allowed_locked) == '15000 NU'
+
+    # Alternate construction
```

Comment:
Not to get too much into the weeds as we mentioned on the call - but it feels like from this line onward is a `unit` test 😊

Reply:
I agree 100% - I think I just missed this one in the great sweep :-)